### DISCUSSION TITLE
Appium-WebDriverAgent (Xcode project) - ld: cannot link directly with 'XCTAutomationSupport' because product being built is not an allowed client of it clang: error: linker command failed with exit code 1 (use -v to see invocation)

### DISCUSSION THREAD

**Discussion by udayapeiris** [appium/appium#20286] [Q&A]
Appium-WebDriverAgent (Xcode project) give the build error for Web~driver Agent `runner
iOS version - 17.5
Appium - 1.22.3
Mac - 14.5
XCODE - 15.4
Appium WebDriver Path - /Applications/Appium\ Server\ GUI.app/Contents/Resources/app/node_modules/appium/node_modules/appium-webdriveragent 

![Screenshot 2024-06-23 at 00 48 07](https://github.com/appium/appium/assets/95946743/adf8f272-92ea-4e21-8c8b-623c0207f5ba)

I tried verious scenarios to find the solution, so for no luck. Can some one please help me

**Comment by KazuCocoa** [comment]
Please check what Xcode version compatibility. https://appium.github.io/appium-xcuitest-driver/latest/installation/requirements/
For Xcode 15 and iOS 17 environments, it requires Appium v2 based xcuitest driver 4.35.0 and more. The latest is recommended.
